tslib/src/functions/check-marker/check-marker.v1.ts

```diff
@@ -0,0 +1,176 @@
+import * as RA from 'ramda-adjunct'
+import * as R from 'ramda'
+import { IAsyncTableService, AsyncTableService } from '../../azure/async-table-service'
+import { ValidatedCheck } from '../../schemas/index'
+import moment from 'moment'
+import { ICheckFormService, CheckFormService } from './check-form.service'
+import { ILogger } from '../../common/ILogger'
+
+export interface ICheckMarkerFunctionBindings {
+  receivedCheckTable: Array<any>
+  checkNotificationQueue: Array<any>
+}
+
+declare interface Answer {
+  factor1: number
+  factor2: number
+  answer: string
+  sequenceNumber: number
+  question: string
+  clientTimestamp: string
+}
+
+declare interface CheckFormQuestion {
+  f1: number
+  f2: number
+}
+
+declare interface MarkingData {
+  answers: Array<Answer>
+  formQuestions: Array<CheckFormQuestion>
+  results: any
+}
+
+declare interface Mark {
+  mark: number
+  maxMarks: number
+  processedAt: Date
+}
+
+export class CheckMarkerV1 {
+
+  private _tableService: IAsyncTableService
+  private _sqlService: ICheckFormService
+
+  constructor (tableService?: IAsyncTableService, sqlService?: ICheckFormService) {
+    if (tableService === undefined) {
+      this._tableService = new AsyncTableService()
+    } else {
+      this._tableService = tableService
+    }
+
+    if (sqlService === undefined) {
+      this._sqlService = new CheckFormService()
+    } else {
+      this._sqlService = sqlService
+    }
+  }
+
+  async mark (functionBindings: ICheckMarkerFunctionBindings, logger: ILogger): Promise<void> {
+
+    const validatedCheck = this.findValidatedCheck(functionBindings.receivedCheckTable)
+    const markingData = await this.validateData(functionBindings, validatedCheck, logger)
+    functionBindings.checkNotificationQueue = []
+    if (markingData === undefined) {
+      functionBindings.checkNotificationQueue.push({
+        checkCode: validatedCheck.RowKey,
+        type: 'unmarkable'
+      })
+      return
+    }
+    const results = this.markCheck(markingData)
+    await this.persistMark(results, validatedCheck)
+    functionBindings.checkNotificationQueue.push({
+      checkCode: validatedCheck.RowKey,
+      type: 'marked'
+    })
+  }
+
+  private async validateData (functionBindings: ICheckMarkerFunctionBindings, validatedCheck: ValidatedCheck, logger: ILogger): Promise<MarkingData | void> {
+    if (RA.isEmptyString(validatedCheck.answers)) {
+      await this.updateReceivedCheckWithMarkingError(validatedCheck, 'answers property not populated')
+      return
+    }
+
+    let parsedAnswersJson: any
+    try {
+      parsedAnswersJson = JSON.parse(validatedCheck.answers)
+    } catch (error) {
+      await this.updateReceivedCheckWithMarkingError(validatedCheck, 'answers data is not valid JSON')
+      return
```

Comment:
I am surprised the preceding `await` was not picked up by a linter.  This is an `async` function so we could 
`return this.updateReceivedCheckWithMarkingError(validatedCheck, 'answers data is not valid JSON')`
and save an `await`.

Am I missing something?

Reply:
i noticed quite a few inconsistencies like that while reviewing the PR yesterday evening.  I will go through the entire project and rectify them